tests/Integration/Adapter/ConfigurationTest.php

```diff
@@ -0,0 +1,415 @@
+        yield [
+            [
+                'key' => 'key_test_1',
+                'value' => 'value_test_1',
+                'shopConstraint' => ShopConstraint::allShops(),
+            ],
+            [
+                'key' => 'key_test_1',
+                'default' => false,
+                'shopConstraint' => ShopConstraint::allShops(),
+            ],
+            'value_test_1',
+        ];
+        // simple case: get a group shop config value
+        yield [
+            [
+                'key' => 'key_test_2',
+                'value' => 'value_test_2',
+                'shopConstraint' => ShopConstraint::shop(2),
+            ],
+            [
+                'key' => 'key_test_2',
+                'default' => false,
+                'shopConstraint' => ShopConstraint::shop(2),
+            ],
+            'value_test_2',
+        ];
+        // simple case: get a single shop config value
+        yield [
+            [
+                'key' => 'key_test_3',
+                'value' => 'value_test_3',
+                'shopConstraint' => ShopConstraint::shopGroup(1),
+            ],
+            [
+                'key' => 'key_test_3',
+                'default' => false,
+                'shopConstraint' => ShopConstraint::shopGroup(1),
+            ],
+            'value_test_3',
+        ];
+        // try to get a non existing value for all shop, get default value instead
+        yield [
+            [],
+            [
+                'key' => 'does_not_exist',
+                'default' => 'default_value_all_shop',
+                'shopConstraint' => ShopConstraint::allShops(),
+            ],
+            'default_value_all_shop',
+        ];
+        // try to get a non existing value for group shop, get default value instead
+        yield [
+            [],
+            [
+                'key' => 'does_not_exist',
+                'default' => 'default_value',
+                'shopConstraint' => ShopConstraint::shopGroup(1),
+            ],
+            'default_value',
+        ];
+        // try to get a non existing value for single shop, get default value instead
+        yield [
+            [],
+            [
+                'key' => 'does_not_exist',
+                'default' => 'default_value',
+                'shopConstraint' => ShopConstraint::shop(2),
+            ],
+            'default_value',
+        ];
+        // get value for a group shop, inherited from all shop
+        yield [
+            [
+                'key' => 'all_shop_key_1',
+                'value' => 'all_shop_value_1',
+                'shopConstraint' => ShopConstraint::allShops(),
+            ],
+            [
+                'key' => 'all_shop_key_1',
+                'default' => false,
+                'shopConstraint' => ShopConstraint::shopGroup(1),
+            ],
+            'all_shop_value_1',
+        ];
+        // get value for shop 2, inherited from parent group shop
```

Comment:
I believe there is one case missing (but I might have missed it)
`// get value for shop 2, inherited from parent group shop` inherited from all shop

And maybe also
`// get value for shop 2, inherited from parent group shop even though there is a value from all shops` (<- group has priority over all shops)